### ISSUE TITLE
[Feature]: ADD CUSTOM LORA ADAPTER CLASS

### ISSUE THREAD

**Issue by Bharathi1604** [vllm-project/vllm#32302]
### 🚀 The feature, motivation and pitch

import torch
import torch.nn as nn
from peft import LoraModel
from transformers import AutoModelForCausalLM, AutoTokenizer

class GatedLoRAModel(nn.Module):
    def __init__(self, base_model_name, lora_config):
        super().__init__()
        # Base frozen GPT OSS model
        self.base_model = AutoModelForCausalLM.from_pretrained(base_model_name)
        self.base_model.requires_grad_(False)

        # LoRA adapted GPT OSS model (trainable)
        self.lora_model = LoraModel(self.base_model, lora_config, adapter_name="default")

        # Gating network: learns how much weight to give LoRA vs base
        hidden_size = self.base_model.config.hidden_size
        self.gate = nn.Sequential(
            nn.Linear(hidden_size, hidden_size // 2),
            nn.ReLU(),
            nn.Linear(hidden_size // 2, 1),
            nn.Sigmoid()  # outputs value in [0,1]
        )

    def forward(self, input_ids, attention_mask=None, labels=None):
        # Hidden states from base (frozen)
        with torch.no_grad():
            base_out = self.base_model(input_ids, attention_mask=attention_mask, output_hidden_states=True)
            hidden_states = base_out.hidden_states  # now this works
        base_logits = base_out.logits
        hidden = base_out.hidden_states[-1].float()  # final hidden states

        # LoRA adapted output
        lora_out = self.lora_model(input_ids, attention_mask=attention_mask)
        lora_logits = lora_out.logits

        # Gate value per token
        gate_vals = self.gate(hidden).squeeze(-1).unsqueeze(-1)  # [batch, seq, 1]

        # Weighted combination of logits
        combined_logits = (1 - gate_vals) * base_logits + gate_vals * lora_logits

        loss = None
        if labels is not None:
            loss_fct = nn.CrossEntropyLoss()
            # shift for causal LM
            loss = loss_fct(
                combined_logits[:, :-1, :].reshape(-1, combined_logits.size(-1)),
                labels[:, 1:].reshape(-1)
            )
        return {"loss": loss, "logits": combined_logits}

If i need to add custom adapter class please provide support as soon as possible. this will  be the best feature

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by jeejeelee** [comment]
Do you mean adding LoRA support for the [router](https://github.com/vllm-project/vllm/blob/v0.13.0/vllm/model_executor/models/gpt_oss.py#L162)?
